design/one-pager-class-selection.md

```diff
@@ -0,0 +1,371 @@
+* _Dynamic provisioning_  is the act of satisfying a _portable_ request for
+  infrastructure (i.e. a resource claim) by automatically provisioning a
+  non-portable managed resource.
+* _Resource classes_ enable dynamic provisioning. A resource class determines
+  what class of managed resource should be created to satisfy a resource claim.
+* _Separation of concern_ happens when an opinionated party can publish classes of
+  infrastructure to be consumed by parties unconcerned with such details.
+  Imagine an organisation in which an infrastructure team supports many product
+  teams who own the lifecycle of their infrastructure. Separation of concerns
+  allows the infrastructure team to define a 'production database' class of
+  service that happens to be a high availability, SSD backed GCP CloudSQL
+  instance. Product teams can then self-service by provisioning a production
+  database and trust that it will be configured appropriately.
+* Crossplane recommends using Kubernetes namespaces to logically group
+  infrastructure into an _infrastructure namespace_, for example
+  `production-infra` or `us-east-coast`. This infrastructure could be claimed
+  and used by applications running in a distinct _application namespace_ such as
+  `widget-service` or `acme-team`.
+
+Note that two forms (classes?) of resource class currently exist; see [portable
+resource classes] for details.
+
+## Goals
+
+The goal of this proposal is to __make it easier for potential Crossplane users
+to learn and take advantage of the portability and separation of concerns__ that
+are enabled by dynamic provisioning. A _secondary_ goal is to maintain the
+flexibility and power that Crossplane offers complex organisations with many
+application and infrastructure namespaces.
+
+## Proposal
+
+This document proposes Crossplane retire the concept of portable resource
+classes. Resource claims would instead be matched directly to non-portable
+resource classes (referred to henceforth as simply 'resource classes') using
+[label selectors].
+
+```yaml
+apiVersion: database.crossplane.io/v1alpha1
+kind: PostgreSQLInstance
+metadata:
+  namespace: acme-team
+  name: gitlab-database
+spec:
+  classSelector:
+    matchLabels:
+      stack: gitlab
+      grade: experimental
+      region: us-east-1
+```
+
+Consider the hypothetical resource claim above, created by an equally
+hypothetical GitLab [stack]. It declares a need for an experimental grade
+PostgreSQL instance in region us-east-1 that is suitable for use with the GitLab
+stack. Any compatible resource class (whether it be an `RDSInstanceClass` or a
+`CloudSQLInstanceClass`) matching the resource claim's labels would be eligible
+to satisfy it. If more than one such class existed one would be selected at
+random. A resource claim that omitted `.spec.classSelector`, `.spec.classRef`,
+and `.spec.resourceRef` would be deemed to be completely unopinionated and would
```

Comment:
That's a good question. I could definitely see the case that if there's an existing, unbound, unblemished managed resource sitting around it could be worth skipping dynamic provisioning and using that managed resource. Perhaps dynamically provisioned managed resources should inherit the labels of their resource classes to enable this at some point?

I agree it's probably best to keep it out of scope for the moment. It would be a forcing function to codify our feelings around https://github.com/crossplaneio/crossplane/issues/539.